src/s3_bench/main.rs

```diff
@@ -0,0 +1,382 @@
+use clap::{command, Arg, ArgAction};
+use object_store::aws::{DynamoCommit, S3ConditionalPut};
+use object_store::path::Path;
+use object_store::ObjectStore;
+use rand::rngs::StdRng;
+use rand::{Rng, RngCore, SeedableRng};
+use slatedb::config::ReadLevel::Uncommitted;
+use slatedb::config::{DbOptions, ObjectStoreCacheOptions, ReadOptions, WriteOptions};
+use slatedb::db::Db;
+use slatedb::inmemory_cache::InMemoryCacheOptions;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use tokio::fs::remove_dir_all;
+use ulid::Ulid;
+
+const SAMPLE_INTERVAL: Duration = Duration::from_secs(1);
+const REPORT_INTERVAL: Duration = Duration::from_millis(100);
+
+#[tokio::main]
+async fn main() {
+    let (params, options, object_store) = configure();
+
+    let db_name = Ulid::new();
+    let s3_db_path = Path::from(format!("s3-bench/{}", db_name));
+    if !params.plot {
+        println!("Creating database: {}", s3_db_path);
+    }
+    let db = Db::open_with_opts(s3_db_path.clone(), options, Arc::clone(&object_store))
+        .await
+        .expect("failed to create database");
+
+    if params.prepopulate_percentage != 0 {
+        if !params.plot {
+            println!(
+                "Prepopulating database with {}% of keys",
+                params.prepopulate_percentage
+            );
+        }
+
+        let mut value = vec![0; params.value_size];
+        rand::thread_rng().fill_bytes(value.as_mut_slice());
+        let seed = rand::thread_rng().next_u32();
+
+        let mut inserted = 0;
+        for key in 0..params.key_count {
```

Comment:
I think we might want the last write to be durable to guarantee that all writes are flushed before we start running the actual benchmark. Otherwise, we might be measuring the tail end of the preload. Not required for this PR, but wanted to make a note of it.

Reply:
could the same be achieved with `db.flush()`?

Reply:
have tried it, fails with `Fenced` status, reverted to a write with `await_durable` set to `true`: https://github.com/slatedb/slatedb/pull/233/commits/33c74cc56ed5158261ec819d25d8e0d3c5595054

Reply:
🤔 That sounds like a bug. Flush should work. I'll open a separate GH issue for that.

Reply:
Opened https://github.com/slatedb/slatedb/issues/235